apps/src/templates/levelSummary/CheckForUnderstanding.jsx

```diff
@@ -0,0 +1,169 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import {ViewType} from '@cdo/apps/code-studio/viewAsRedux';
+import getScriptData from '@cdo/apps/util/getScriptData';
+import SafeMarkdown from '@cdo/apps/templates/SafeMarkdown';
+import SectionSelector from '@cdo/apps/code-studio/components/progress/SectionSelector';
+import i18n from '@cdo/locale';
+import styles from './check-for-understanding.module.scss';
+
+const SUMMARY_PARAM = 'view=summary';
+const FREE_RESPONSE = 'FreeResponse';
+
+const CheckForUnderstanding = ({
+  isRtl,
+  viewAs,
+  selectedSection,
+  students,
+  currentLevelId,
+  levels
+}) => {
+  const currentLevel = levels.find(l => l.activeId === currentLevelId);
+  const nextLevel = levels.find(l => l.position === currentLevel.position + 1);
+
+  // To avoid confusion, if a teacher tries to view the summary as a student,
+  // send them back to the level in Participant mode instead.
+  if (viewAs === ViewType.Participant) {
+    const paramString = document.location.search
+      .replace(SUMMARY_PARAM, '')
+      .replace('&&', '&')
+      .replace('?&', '?');
+    document.location.replace(currentLevel.url + paramString);
+  }
+
+  const data = getScriptData('summary');
+
+  const questionMarkdown = data.level.properties.long_instructions;
+  const teacherMarkdown = data.teacher_markdown;
+  const height = data.level.height || '80';
+
+  return (
+    <div className={styles.summaryContainer}>
+      {/* Top Nav Links */}
+      <p className={styles.navLinks}>
+        <a href={currentLevel.url}>&lt; {i18n.backToLevel()}</a>
+        {nextLevel && (
+          <a
+            className={isRtl ? styles.navLinkLeft : styles.navLinkRight}
+            href={`${nextLevel.url}${document.location.search}`}
+          >
+            ${i18n.nextLevelLink()} &gt;
+          </a>
+        )}
+      </p>
+
+      {/* Question Title */}
+      {data.level.properties.title && (
+        <h1 className={styles.levelTitle}>{data.level.properties.title}</h1>
+      )}
+
+      {/* Question Body */}
+      <SafeMarkdown className={styles.markdown} markdown={questionMarkdown} />
+
+      {/* Question Inputs */}
+      {data.level.type === FREE_RESPONSE && (
+        <textarea
+          className={styles.freeResponse}
+          id={`level_${data.level.id}`}
+          aria-label={i18n.yourAnswer()}
+          placeholder={
+            data.level.properties.placeholder || i18n.enterYourAnswerHere()
```

Comment:
i could reuse this for the label, but I didn't think it made as much sense, since it's intended for the placeholder. this could be weird if curriculum designers write something important as a placeholder though 🙃